packages/frontend/features/common/useToast.ts

```diff
@@ -0,0 +1,23 @@
+export const useToastShowLoading = (tx, action: string) => {
+  window.toastProvider.addMessage(`${action.charAt(0).toUpperCase() + action.substring(1)}...`, {
+    secondaryMessage: "Check progress on Etherscan",
+    actionHref: `https://etherscan.io/tx/${tx.hash}`,
+    actionText: "Check",
+    variant: "processing",
+  });
+};
+
+export const useToastHandleException = (tx, action: string) => {
```

Comment:
the `useFoo` conventions is only for React hooks. It might not be a good idea to name it `useFoo` because there are React linters that pick up on this convention and will error out assuming this is a hook.

Reply:
I would recommend that `useToastShowLoading` and `useToastHandleException` be changed to `showTxLoadingToast` and `handleTxException`.

Reasons:

- removes React hook convention of `useFoo`
- allows `handleTxException` to be more extensible (e.g. adding analytics reporting for people who cancel)

I also recommend this file be changed to `txUtils.ts` as they don't have to be toast specific but they are common utils that we need when handling transactions.